### PR TITLE
Bumped media player to version 2.21.2

### PR DESCRIPTION
**Pull request by brian-learningpool**
This corrects issues with YouTube videos losing their controls on Firefox and IE 11.

### CODE REVIEW

**Comment by barry-learningpool** [comment]
+1

**Comment by moloko** [comment]
Sorry but this update completely breaks Adapt for me, in every desktop browser I've tried... this is the error I get in Chrome, though I'm not sure it's that helpful...
media-pr-143-error
If I compare this version to the current master branch, I can see that master has quite a lot more code in mediaelement-and-player.js ... basically everything from [line 2067](https://github.com/adaptlearning/adapt-contrib-media/blob/master/js/mediaelement-and-player.js#L2067) onwards isn't present in the updated version.

**Comment by brian-learningpool** [comment]
Thanks, @moloko -- good catch!  Copy and paste gone wrong on the previous commit by the looks of it.

**Comment by brian-learningpool** [comment]
I'm seeing ARIA highlighting now on the player controls when tested on a course with the Vanilla theme though.

**Comment by moloko** [comment]
the play button did previously get a yellow border around it sometimes - so if that's what you're referring to, that's not something this update has introduced

**Comment by moloko** [comment]
I'd also get aria-highlighting on the controls when clicking them, in Firefox anyway. Again - not something this update has introduced.

**Comment by moloko** [comment]
8c68cd6 has definitely got it all back up and running again and all seems fine in Firefox. I'd like to do some more cross-browser testing if that's OK but I've run out of time today and am not working tomorrow. Can this wait until Monday? If not, it's probably fine I'm just being ultra-cautious ;-) A quick test by someone in IE8 before merging would probably be a good plan though.

**Comment by oliverfoster** [comment]
Seconded. Ie9 and 10 are always fun with video also. We need to check that they mark as complete on play and finished from ie8-edge

**Comment by brian-learningpool** [comment]
Browser testing:

- [ ] IE 8
- [ ] IE 9
- [ ] IE 10
- [ ] IE 11
- [ ] Edge

**Comment by moloko** [comment]
In Firefox, if you have closed captions enabled with a default language set, they don't appear until you switch them on. In Edge, IE11 and Chrome they appear automatically. Quite possibly a pre-existing issue, need to check.

**Comment by moloko** [comment]
things I've been testing:
* fullscreen video (except in IE9 which doesn't support it)
* closed-captions
* youtube & vimeo
* half-width component layout
* all three completion criteria
* that simultaneous playback of multiple players is not possible (except with Vimeo)
* keyboard accessibility (IE only)

**Comment by moloko** [comment]
IE8 - script errors generated, originating from the Vimeo player. 

2016-06-27 15_50_46-ie8 - win7 original state running - oracle vm virtualbox

need to check if pre-existing issue or not

**Comment by moloko** [comment]
IE8 - player control bar doesn't appear when source is YouTube

Need to check if pre-existing issue or not

**Comment by brian-learningpool** [comment]
IE 9 - completion on play of YouTube videos doesn't appear to work.

**Comment by brian-learningpool** [comment]
@moloko, have you noticed YouTube content looping once it's finished playing?

**Comment by moloko** [comment]
@brian-learningpool Checked in IE11, Firefox and Chrome - no looping for me. Only oddity I noticed was that in Firefox it showed a 'loading spinner' once the video had ended:
2016-07-08 15_09_31-adapt version 2 0 demonstration _ presentation components

**Comment by brian-learningpool** [comment]
I'm seeing the YouTube content looping on iOS.

**Comment by moloko** [comment]
it's fine for me on iPad Mini/iOS 9.3.2 - just going to update to iOS 9.3.3 and try again

**Comment by moloko** [comment]
OK so weirdly the looping thing has now stopped occurring for me in Firefox (no idea why) and I now get the same behaviour as for Chrome. It's still happening in IE11 & Edge though.

Also now tested on iPad Mini/iOS 9.3.3 - it happened there too initially, but as with Firefox it then started working.

I've also noticed that I get pretty much the same results when testing the [YouTube API example here](http://mediaelementjs.com/examples/?name=youtube) so this strongly suggests it's a bug in MediaElement itself (also note that's the most recent release: v2.22.0)

File: js/mediaelement-and-player.js [hunk 1]
```diff
@@ -680,7 +686,9 @@ mejs.PluginMediaElement.prototype = {
 	pause: function () {
 		if (this.pluginApi != null) {
 			if (this.pluginType == 'youtube' || this.pluginType == 'vimeo') {
-				this.pluginApi.pauseVideo();
+		        if( this.pluginApi.getPlayerState() == 1 ) {
```
Comment (moloko) [inline js/mediaelement-and-player.js]: it's this `if` statement that's causing the looping issue

**Comment by moloko** [comment]
https://github.com/johndyer/mediaelement/issues/1795

File: js/mediaelement-and-player.js (hunk 1 above)
Comment (brian-learningpool) [inline js/mediaelement-and-player.js]: Great find @moloko -- that resolves it for me also.

**Comment by brian-learningpool** [comment]
Hi @moloko, this has been open for a while.  Do you think we should commit with your fix mentioned earlier in this thread?

File: js/mediaelement-and-player.js [hunk 2]
```diff
@@ -685,14 +685,12 @@ mejs.PluginMediaElement.prototype = {
 	},
 	pause: function () {
 		if (this.pluginApi != null) {
+			// PATCH -- Prevent looping video on Vimeo and YouTube
```
Comment (moloko) [inline js/mediaelement-and-player.js]: could you add link to issue in the comment? 

https://github.com/johndyer/mediaelement/issues/1795

File: properties.schema [hunk 3]
```diff
@@ -92,7 +92,7 @@
           "title": "Poster Link",
           "inputType": "Asset:image",
           "validators": [],
-          "help": "Image displayed before the video is played"
+          "help": "Image displayed before the video is played (except oh Vimeo)"
```
Comment (moloko) [inline properties.schema]: typo: 'oh' should be 'on' I presume?

Comment (brian-learningpool) [inline properties.schema]: Probably a Freudian slip at this stage!  Good catch.

**Comment by moloko** [comment]
+1

**Comment by oliverfoster** [comment]
+1

**Comment by brian-learningpool** [comment]
Merging to `develop` for a full browser run through before merging to `master`.

**Pull request merged**